### ISSUE TITLE
undefined method 'filter'

### ISSUE THREAD

**Issue by ghost** [sous-chefs/line#140]
### Cookbook version
2.2.0

### Chef-client version
14.10.9

### Platform Details
RedHat 7

### Scenario:
Run basic filter_lines

### Steps to Reproduce:
depends 'line' # put in metadata.rb
RECIPE:
content = <<~EOF
  export var=EXAMPLE
EOF
match_pattern = /System wide functions and aliases/
filter_lines 'insert var' do
  path '/etc/bashrc'
  filter after: [match_pattern, content]
end

### Expected Result:
Success

### Actual Result:
NoMethodError: undefined method 'filter' for Custom resource line_filter_files from cookbook line

**Comment by MarkGibbons** [comment]
Could you try that with the filters instead of filter property?  There's a typo in the documentation examples.

**Comment by ghost** [comment]
Thanks that was it. I added a not_if { ::File.read("/etc/bashrc").include?(content) } and it does what I want.

**Comment by MarkGibbons** [comment]
If you need to add that not_if something is wrong.  Could you tell me what happened without the not_if?

Thanks

**Comment by ghost** [comment]
It simply matches the line and adds my content again. I assumed that was expected behavior. Is there another way to construct the recipe to prevent that?

**Comment by MarkGibbons** [comment]
content='export var=EXAMPLE' . The EOL characters are unexpected (dumb on my part).  

or
content = << -EOF
  line1
  line2 
EOF.split("\n")

Filters expect strings without EOL, or an array of strings without EOL characters.

**Comment by ghost** [comment]
Your example doesn't work due to the 'EOF' must be on it's own line. This works:
```
content = <<~EOF
  export foo=bar
  export bar=baz
EOF
content = content.split("\n")
```
I prefer the squiggle HEREDOC available since Ruby 2.3 (https://infinum.co/the-capsized-eight/multiline-strings-ruby-2-3-0-the-squiggly-heredoc) as it allows me to indent but strips the leading whitespace.

**Comment by MarkGibbons** [comment]
A fix for your original input is in the works.

**Comment by MarkGibbons** [comment]
The documentation errors and allowing lines to be specified in a string have been fixed by #141.  Splitting the lines into an array before handing them to the sample filters is no longer required.
